### DISCUSSION TITLE
Request a firmware build - 2.7.0

### DISCUSSION THREAD

**Discussion by pfeerick** [EdgeTX/edgetx#1836] [General]
Ask here if you want/need specific options that are not included in the stock builds. Builds will be uploaded to [here (same link as 'By request' below)](http://files.portabledev.xyz/d/859f9d6867e24b80b5f0/), so check it out before making a request just in case there is already a custom build with your chosen options...

*Please remember to specify your transmitter when requesting a build* ;)

- [Multilingual builds](http://files.portabledev.xyz/d/66a3a1f8d8dd42c48888/) - zip file per transmitter containing all languages built
- [Multilingual builds - NOHELI](http://files.portabledev.xyz/d/ab71d9948d4941b5be90/) - zip file per transmitter containing all languages built
- [By request](http://files.portabledev.xyz/d/859f9d6867e24b80b5f0/)

**Comment by ArdSchell** [comment]
Can you build an EdgeTX 2.7 with bluetooth enabled? 
Thanks!!

**Reply by tedsears** [reply]
This would be great.

**Reply by pfeerick** [reply]
Done! :) Check the by request link in the first post. 
`cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DBLUETOOTH=Y DCMAKE_BUILD_TYPE=Release ..`
`cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DHELI=N -DBLUETOOTH=Y DCMAKE_BUILD_TYPE=Release ..`

**Reply by pfeerick** [reply]
Assuming, of course, this is for the TX16S? ;)

**Reply by tedsears** [reply]
Tx16s for me at least!  Thanks so much!

**Reply by ArdSchell** [reply]
Thanks! This is great. :O)

**Reply by ArdSchell** [reply]
Yes, sorry. Its for the TX16S. Thanks!!

**Comment by SimitarKnut** [comment]
I would like to see on the model set up, the choice between electric or fuel, motor/engine. Then on the mix page, instead of doing this otherwise, have the option to assign a switch to mix with the throttle to:

If electric, override the throttle to cut off.

If fuel, if throttle is below a certain position to cut the throttle,  but if above a certain position, but below half, (or a programmable weight range) to hold the throttle instead of cutting it.

Thanks to all you good people who are doing your best to make our hobby better!

Claude
AKA Simitar Knut

**Reply by pfeerick** [reply]
Sorry, this is the wrong place to be posting this... can you bring it up as a seperate post in [ideas](https://github.com/EdgeTX/edgetx/discussions/categories/ideas) rather than in the firmware request thread thanks! :) 

Having said that, this sounds like something that could be done with a template - i.e. have a template pre-configured for fuel motors with logical switches to do that task?

**Comment by wertoflc** [comment]
I have TX16s with frsky R9M access module .Can you build me firmware with also noheli option? Thank you so much !

**Reply by pfeerick** [reply]
Done. :) Check the by request link in the first post.
`cmake -Wno-dev -DPCB=X10 -DPCBREV=TX16S -DHELI=N -DHARDWARE_EXTERNAL_ACCESS_MOD=ON DCMAKE_BUILD_TYPE=Release ..`

**Reply by wertoflc** [reply]
Thank you so much :) Worked like a charm

**Comment by enzo-molinari** [comment]
Hello

I have an Horus X12S with touchscreen upgrade.Can you build me firmware with also touchscreen option and with bluetooth enabled? Thank you so much!

Enzo

**Reply by pfeerick** [reply]
Is this with the https://github.com/EdgeTX/edgetx/pull/1779 TX16S touchscreen mod? This won't be in a release version of EdgeTX until 2.8.0. You should be able to use a nightly build in the meantime as both bluetooth and touchscreen  support are enabled by default.

**Reply by enzo-molinari** [reply]
Peter

Yes, it is the #1779 TX16S touchscreen mod?
Thank you for the reply and thank you for what you are doing for the EdgeTx community.

Kind regards Enzo

**Comment by jeffhendrix** [comment]
T16 with bluetooth (I already updated my TX16S, I like to keep my radios in sync)
Also, does the X9D+ build work on the older X9D?

**Reply by pfeerick** [reply]
Done. As before, check the links in the first post. 

`cmake -Wno-dev -DPCB=X10 -DPCBREV=T16 -DINTERNAL_MODULE_MULTI=ON -DBLUETOOTH=Y ../`

> Also, does the X9D+ build work on the older X9D?

That particular one won't, since it is for the X9D+2019, which has a completely different internal RF... 

`cmake -Wno-dev -DPCB=X9D -DFONT=SQT5 -DHELI=NO ../`

**Comment by Kikdiweg** [comment]
Hi,
Xlite ( non s/Pro) internal module is not working and found that FIRMWARE OPTIONS in radio says "internalaccess" "internalmulti". Can you build one with correct internal module specified

**Reply by pfeerick** [reply]
What is `SYS->HARDWARE->Internal RF Type`set to? Is it XJT or ISRM? It looks like the XLITE and XLITES builds are correctly configuring the _default_ internal module as XJT or ISRM respectively, so it is probably the user level setting that is overriding this.

**Comment by CorruptName** [comment]
Can I possibly get:
`noheli + externalaccessmod + bluetooth`

for TX16S please?

**Reply by pfeerick** [reply]
I'll do this today. I've just been holding off as 2.7.1 was just around the corner, so will be doing a build for that instead.

**Reply by pfeerick** [reply]
```cmake -Wno-dev -DCMAKE_BUILD_TYPE=Release -DPCB=X10 -DPCBREV=TX16S  -DHARDWARE_EXTERNAL_ACCESS_MOD=ON -DHELI=N -DBLUETOOTH=Y ..```

Use the by request link in #1951 since it is a 2.7.1 build

**Comment by rcman45** [comment]
Bonjour
Je voudrais un firmware 2.7.1 FrSky X12S avec EXTERNAL-ACCESS pour R9 + TOUCH-SCREEN  et en francais (FR)
Merci

**Reply by pfeerick** [reply]
```cmake -Wno-dev -DCMAKE_BUILD_TYPE=Release -DPCB=X12S -DHARDWARE_EXTERNAL_ACCESS_MOD=ON -DTRANSLATIONS=FR .. ```

Touchscreen is automatic if installed.

Use the by request link in https://github.com/EdgeTX/edgetx/discussions/1951 since it is a 2.7.1 build
